batcher/aligned-sdk/src/core/errors.rs

```diff
@@ -195,8 +195,12 @@ impl fmt::Display for SubmitError {
             SubmitError::InvalidProof(reason) => write!(f, "Invalid proof {}", reason),
             SubmitError::ProofTooLarge => write!(f, "Proof too Large"),
             SubmitError::InvalidReplacementMessage => write!(f, "Invalid replacement message"),
-            SubmitError::InsufficientBalance(addr) => {
-                write!(f, "Insufficient balance, address: {}", addr)
+            SubmitError::InsufficientBalance(addr, last_sent_valid_nonce) => {
+                write!(
+                    f,
+                    "Insufficient balance, address: {} last_sent_valid_nonce: {}",
```

Comment:
I think sending the `last_valid_sent_nonce` is unnecessary and over complicated.
Sending the `nonce with InsufficientBalance` is much more straight forward